src/triggers/action_performed.ts

```diff
@@ -11,12 +11,38 @@ function getActionPerformance(z: ZObject, bundle: Bundle) {
     return [bundle.cleanedRequest.data]
 }
 
-async function getFallbackRealActionPerformance(z: ZObject, bundle: Bundle) {
+export async function getFallbackRealActionPerformance(z: ZObject, bundle: Bundle) {
     const action_id = bundle.inputData.action_id
-    const response = await z.request({
-        url: composeUrl(['api', 'projects', bundle.inputData.project_id, 'events', `?action_id=${action_id}`], bundle),
+    const eventsResponse = await z.request({
+        url: composeUrl(['api', 'projects', bundle.inputData.project_id, 'query'], bundle),
+        method: 'POST',
+        body: {
+            query: {
+                kind: 'EventsQuery',
+                select: ['*', 'person'],
+                actionId: action_id,
+                after: 'all',
+                limit: 3,
+                orderBy: ['timestamp DESC'],
+            },
+        },
     })
-    return (response.data as { results: Record<string, any>[] }).results
+    // we need to transform the response from the events query to look like
+    // the one from an actual "rest webhook" call
+    const response = (eventsResponse.data.results as Record<string, any>[][]).map((row) => {
+        const { uuid, team_id, distinct_id, elements, elements_chain: _unused, ...event } = row[0]
+        const person = row[1]
+        return {
+            eventUuid: uuid,
+            teamId: team_id,
+            distinctId: distinct_id,
+            elementsList: elements,
+            ...event,
+            person,
+        } as Record<string, any>
+    })
```

Comment:
> Feels like a waste of the query node's capabilities to do the processing here, instead of selecting just uuid AS id, distinct_id, properties, event, timestamp, and person in the select array.

My reasoning for `select *` is that any future additions will be included without changes here (where they are easy to forget).

> BTW why does the new version transform uuid to eventUuid and not the previous id? I don't think this was the case before, since the currently live version of the integration just uses results from the /events/ endpoint verbatim

Since the output of this call is used to select properties for filtering etc. we want this to resemble the actual webhooks call as much as possible. In the actual call we send an `eventUuid` and not an `id`. See [this test](https://github.com/PostHog/posthog/blob/master/plugin-server/functional_tests/webhooks.test.ts#L182-L222) or the src/test/__mocks__/rest_webhook.json file, which is the output of an actual webhook call.

Reply:
Ohh, right I missed that the plugin server does things slightly differently.

As for future additions, I think them being omitted by default would be a _feature_, given we likely want this API to remain stable regardless of future additions to HogQL (which likely won't be reflected in REST hooks)